src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs

```diff
@@ -1,14 +1,32 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.MSBuild;
 using Microsoft.VisualStudio.ComponentModelHost;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
 using Microsoft.VisualStudio.Shell.Interop;
+using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
 {
     internal partial class VisualStudioProjectTracker : IVsSolutionLoadEvents
     {
+        private CancellationTokenSource _solutionParsingCancellationTokenSource = new CancellationTokenSource();
+
         int IVsSolutionLoadEvents.OnBeforeOpenSolution(string pszSolutionFilename)
         {
+            Task.Delay(TimeSpan.FromSeconds(10), _solutionParsingCancellationTokenSource.Token)
+                .ContinueWith(
+                    _ => LoadSolutionInBackground(pszSolutionFilename),
+                    _solutionParsingCancellationTokenSource.Token,
```

Comment:
don't you want to capture the token at the begining of this method since _solutionParsingCancellationTokenSource is mutable.